src/order/create-order.ts

```diff
@@ -56,24 +57,10 @@ export default async function createOrder(
       body: JSON.stringify(orderPayload),
     });
 
-    const data = await response.json();
+    const data = await response.json() as CreateOrderResponse;
 
-    if (response.status !== 200 && response.status !== 201) {
-      const errorData = data as CreateOrderErrorResponse;
+    return { data, httpStatus:response.status} as {data: CreateOrderResponse, httpStatus:number };
```

Comment:
Wondering if we can remove the casting here. Assuming TS will know the types for these.
```suggestion
    return { data, httpStatus:response.status};